include/nuttx/queue.h

```diff
@@ -107,6 +107,17 @@
     } \
   while (0)
 
+#define dq_addfirst_notempty(p, q) \
```

Comment:
Do not expose this type of API to users, this will confuse developers who read the API and cause serious issues if misused. you can implement it  in your code privately for performance version